checkmates/data_checks/checks/distribution_data_check.py

```diff
@@ -0,0 +1,157 @@
+"""Data check that checks if the target data contains certain distributions that may need to be transformed prior training to improve model performance."""
+import diptest
+import numpy as np
+import woodwork as ww
+
+from checkmates.data_checks import (
+    DataCheck,
+    DataCheckActionCode,
+    DataCheckActionOption,
+    DataCheckError,
+    DataCheckMessageCode,
+    DataCheckWarning,
+)
+from checkmates.utils import infer_feature_types
+
+
+class DistributionDataCheck(DataCheck):
+    """Check if the target data contains certain distributions that may need to be transformed prior training to improve model performance. Uses the skew test and yeojohnson transformation."""
+
+    def validate(self, X, y):
+        """Check if the target data has a skewed or bimodal distribution.
```

Comment:
Should be changing this text since this checks the features (and maybe target as well)

Reply:
adjusted docstring and code to reflect checking features and target